### PR TITLE
Fix Stream and Event nullptr value

### LINKED ISSUE #793: Stream object should have .ptr set to 0, not None.
The event object expects the stream.ptr to be an integer (size_t) here:
https://github.com/pfnet/chainer/blob/master/cupy/cuda/stream.py#L56
https://github.com/pfnet/chainer/blob/master/cupy/cuda/runtime.pyx#L309

In trunk at the moment, recording events with default stream fails via:
Traceback (most recent call last):
  File "train_imagenet.py", line 85, in <module>
    train_loop()
  File "train_imagenet.py", line 67, in train_loop
    start.record()
  File "/home/awesomebox/anaconda/lib/python2.7/site-packages/chainer-1.5.1-py2.7-linux-x86_64.egg/cupy/cuda/stream.py", line 56, in record
    runtime.eventRecord(self.ptr, stream.ptr)
  File "cupy/cuda/runtime.pyx", line 309, in cupy.cuda.runtime.eventRecord (cupy/cuda/runtime.cpp:6139)
TypeError: an integer is required

The fix seems simple:

https://github.com/pfnet/chainer/blob/master/cupy/cuda/stream.py#L103
self.ptr = 0

### PR DESCRIPTION
**Pull request by beam2d**
Fix #793. It still lacks tests (as is).

### CODE REVIEW

File: cupy/cuda/stream.py [hunk 1]
```diff
@@ -22,7 +22,7 @@ class Event(object):
 
     """
     def __init__(self, block=False, disable_timing=False, interprocess=False):
-        self.ptr = None
+        self.ptr = 0
```
Comment (delta2323) [inline cupy/cuda/stream.py]: `self.ptr` is set at the last line of `__init__`.  So I guess it is not necessary to initialize `self.ptr` here.

File: cupy/cuda/stream.py [hunk 2]
```diff
@@ -109,7 +109,7 @@ def __init__(self, null=False, non_blocking=False):
     def __del__(self):
         if self.ptr:
             runtime.streamDestroy(self.ptr)
-            self.ptr = None
+            self.ptr = 0
```
Comment (delta2323) [inline cupy/cuda/stream.py]: Do we have to set `self.ptr` to 0 explicitly even if `Stream` object itself will be deleted at the same time?

File: cupy/cuda/stream.py (hunk 1 above)
Comment (beam2d) [inline cupy/cuda/stream.py]: It is indeed required for the case that __init__ raises an exception before setting the ptr. If we omit the first line, __del__ tries to access ptr, and raises an AttributeError. This is problematic.

Comment (delta2323) [inline cupy/cuda/stream.py]: OK, I understand it.

**Comment by delta2323** [comment]
LGTM :+1:

**Pull request merged**